strider/fetcher.py

```diff
@@ -257,26 +263,34 @@ def get_kp_request_body(
         kp: dict,
 ) -> Response:
     """Get request to send to KP."""
-    included_nodes = [
-        qgraph['edges'][step.edge]['subject'],
-        qgraph['edges'][step.edge]['object'],
-    ]
-    included_edges = [step.edge]
 
+    predicate, _ = \
+        extract_predicate_direction(kp["edge_predicate"])
+
+    # Build request edges
+    request_edge = qgraph["edges"][step.edge].copy()
+    request_source = qgraph["nodes"][step.source].copy()
+    request_target = qgraph["nodes"][step.target].copy()
+
+    # Update request properties to match what the KP expects
+    request_edge["subject"] = step.source
```

Comment:
It makes me nervous any time we have "subject" and "source", or "object" and "target", in the same line. The logic here is not obvious.